### PR TITLE
use context to create new loggers

### PR DESCRIPTION
**Pull request by KPostOffice**
# Issue link
https://github.com/project-codeflare/codeflare-operator/issues/113

# What changes have been made
Use the controller runtime logger everywhere with structured logging. The logger is generated based on the current context object. In the linked issue there is talk about using `.WithName` but I'm unsure if this is necessary.

# Verification steps
Run controller and see new structured logs.

## Checks
- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [ ] Testing is not required for this change

### CODE REVIEW

File: controllers/machinepools.go [hunk 1]
```diff
@@ -107,9 +118,15 @@ func (r *AppWrapperReconciler) deleteMachinePool(ctx context.Context, aw *arbv1.
...
+				logger.Error(
+					err, "Error deleting machinepool",
+					"machinePool", targetMachinePool,
+				)
 			}
-			klog.Infof("Successfully Scaled down target machinepool %v", id)
+			logger.Info(
+				"Successfully scaled down target machinepool",
+				"machinePool", targetMachinePool,
+			)
```
[the remark this answers is not in the document]
Comment (KPostOffice) [reply controllers/machinepools.go:129]: Would returning the error be enough?

[the remark this answers is not in the document]
Comment (VanillaSpoon) [reply controllers/machinepools.go:129]: Hey @KPostOffice ,
I believe so, or actually even an else would be sufficient, and would prevent the error from reconciling over and over?
```
else {
	logger.Info(
				"Successfully scaled down target machinepool",
				"machinePool", targetMachinePool,
			)
}
```

**Pull request merged**